### ISSUE TITLE
Exits to main menu on failed RTMP connection

### ISSUE THREAD

**Issue by fragtion** [xbmc/xbmc#22186]
### Describe the bug

If a playlist or .strm file contains a link to an RTMP stream (eg rtmp://someurl/live/stream), and the stream is inaccessible (or the connection drops during playback), the playlist is aborted and kodi crashes to the main menu with ShowOKDialogText {16026}/{16027} (_"One or more items failed to play. Check the log for more information about this message"_).
If the stream was already playing, and the connection drops, it seems to try to reconnect once only, then gives up.

Setting PlayerControl to Repeat or RepeatAll, does not solve the problem.
Setting high timeout values for **playlistretries** and **playlisttimeout** in advancedsettings.xml, also has no effect

Problem is also described by other users in some threads, without any solution thus far - eg:
https://forum.kodi.tv/showthread.php?tid=248895
https://forum.kodi.tv/showthread.php?tid=230435
https://forum.kodi.tv/showthread.php?tid=308113
https://www.reddit.com/r/Addons4Kodi/comments/hiwx95/automatically_restart_a_dropped_stream/

## Expected Behavior

The unplayable item should soft-fail, skipping to the next item in the playlist.
It should also retry connection according to **playlistretries** count, and respect **playlisttimeout** as expected, which it currently does not. These options seem to be ignored completely even when using an m3u playlist.

## Possible Fix

I presume this has something to do with player logic in [PlayListPlayer.cpp], but unfortunately I do not know for sure.

### Your Environment

Occurs on all playforms I've tested (x86/windows & LibreELEC media box)

 - Operating system version/name:
 - Kodi version:
Latest

**Comment by lrusak** [comment]
If you could post a backtrace that would be helpful, this is generated automatically on LibreELEC. You can access it at `~/.kodi/temp/kodi_crash.log`

**Comment by fragtion** [comment]
Ah.. I think maybe I used the wrong term saying it "crashes", as it seems to just be falling back to the startup screen the 16026 OK dialog as I described... But that seems like a good idea anyway so I'll check for a crash log and revert back soon

**Comment by lrusak** [comment]
> Ah.. I think maybe I used the wrong term saying it "crashes", as it seems to just be falling back to the startup screen with the 16026 "OK Dialog box" as I already described... But the crash log sounds like a good idea so I'll check for that anyway & post an update on that soon

If it's not crashing the a debug log may help anyways. The debug log is located in the same folder as the crash log.

**Comment by a1rwulf** [comment]
Yeah I think this is an issue since quite some time, iirc around XBMC Frodo it was possible to automatically try the next item in an m3u, but apparently that got lost since then.
That being said, imho it's not trivial to fix this correctly for all protocols - I looked into this in the past, but never had enough time to really get something going.
